### PR TITLE
React calendar data renderer test

### PR DESCRIPTION
**Pull request by noahsug**
The withDisplayRenderer() wrapper makes testing a bit painful. I could export a direct reference to <CalendarInput> used by tests only, but adding test only things is lame.

TODO: 
 - add tests for 'date' and 'time' only modes
 - add tests for date picker interaction

### CODE REVIEW

**Comment by DeadHeadRussell** [comment]
To alleviate the `withDataRenderer` wrapper, we could move the component related code to its own file in components (eg, `./src/react/components/calendar.jsx`) and leave the renderers code to import it and focus solely on what wrappers for how it is being displayed.

File: src/react/renderers/data/index.jsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 export function withDataRenderer(WrappedComponent) {
   return ({viewType, renderData}) => {
-    const {getError, isDisabled, onChange, onBlur, nButtonClick} = renderData.options;
+    const {getError, isDisabled, onChange, onBlur, onButtonClick} = renderData.options;
```
Comment (DeadHeadRussell) [inline src/react/renderers/data/index.jsx]: And this is why we need tests! :-o

**Comment by noahsug** [comment]
Moving the component code into react/components is a good idea, but I think it's okay to leave as is since the extra work is already done.

I added another test to ensure taking input from the picker works. If this looks good to you, I think we can submit this!

**Comment by DeadHeadRussell** [comment]
Yup, sounds good.

**Pull request merged**